internal/subshell/subshell.go

```diff
@@ -52,7 +54,7 @@ type SubShell interface {
 	SetBinary(string)
 
 	// WriteUserEnv writes the given env map to the users environment
-	WriteUserEnv(sscommon.Configurable, map[string]string, sscommon.RcIdentification, bool) error
+	WriteUserEnv(sscommon.Configurable, map[string]string, sscommon.RcIdentification, bool, bool) error
```

Comment:
I think the bool flag "create" here has the wrong connotation. The logic that this is facilitating is "only if the user has it". I'd rather see the implementation convey that logic.

Additionally, you are still calling WriteRCFile() even when create=false, which is banking on WriteRCFile() implicitly handing this scenario.